rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -3141,6 +3158,46 @@ public void addNodeToken(String nodeUrl, String token) throws RMException {
         }
     }
 
+    @Override
+    public Set<String> getNodeTags(String nodeUrl) throws RMException {
+        if (allNodes.containsKey(nodeUrl)) {
+            RMNode rmNode = allNodes.get(nodeUrl);
+            return rmNode.getNodeTags();
+        } else {
+            throw new RMException("Unknown node: " + nodeUrl);
+        }
+    }
+
+    @Override
+    public Set<String> getNodesByTags(String tag) {
```

Comment:
Wondering if an exception should be throw when a tag is not found

Reply:
It will simply return an empty list in this case.

Reply:
the method should be called getNodesByTag